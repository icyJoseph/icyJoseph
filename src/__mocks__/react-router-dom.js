import React, { Fragment } from "react";

const rrd = require("react-router-dom");

rrd.BrowserRouter = ({ children }) => <Fragment>{children}</Fragment>;

module.exports = rrd;
