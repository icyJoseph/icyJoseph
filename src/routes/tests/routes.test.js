import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import Routes, {
  AsyncLanding,
  AsyncHacks,
  AsyncBlog,
  AsyncNoMatch,
  AsyncTopMenu
} from "..";

// jest.mock("react", () => {
//   const r = jest.requireActual("react");

//   return { ...r, memo: x => x };
// });

describe("SnapShot Routes", () => {
  const wrapper = shallow(<Routes />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("AsyncLanding", () => {
  const loaded = shallow(<AsyncLanding />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncHacks", () => {
  const loaded = shallow(<AsyncHacks />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncBlog", () => {
  const loaded = shallow(<AsyncBlog />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncNoMatch", () => {
  const loaded = shallow(<AsyncNoMatch />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("AsyncTopMenu", () => {
  const loaded = shallow(<AsyncTopMenu />);
  it("returns", () => {
    expect(loaded.exists()).toEqual(true);
  });
});

describe("Routes", () => {
  const mockedStore = {
    getState: jest.fn(() => ({
      github: { repos: [] },
      drawer: { open: false, id: 0 },
      medium: { feed: { articles: {}, user: {} } }
    })),
    dispatch: jest.fn(),
    subscribe: jest.fn()
  };

  it("renders for blog", () => {
    const wrapper = mount(
      <Provider store={mockedStore}>
        <MemoryRouter initialEntries={["/blog"]}>
          <Routes />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find(AsyncBlog)).toHaveLength(1);
    expect(wrapper.find(AsyncLanding)).toHaveLength(0);
  });
});
