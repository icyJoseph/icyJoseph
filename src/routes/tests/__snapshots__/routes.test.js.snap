// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot Routes renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "basename": "",
      "children": <React.Fragment>
        <GlobalStyleComponent />
        <Route
          component={[Function]}
          path="/:activeItem?"
        />
        <ForwardRef>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/hacks"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/blog"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </ForwardRef>
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GlobalStyleComponent />,
          <Route
            component={[Function]}
            path="/:activeItem?"
          />,
          <ForwardRef>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/hacks"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/blog"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/:activeItem?",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/hacks"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/blog"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/hacks"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/blog"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/hacks",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/blog",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex: 1;

  > div:first-child {
    display: flex;
    flex-direction: column;
    width: 100%;
  }

  > div > div:last-child {
    margin-bottom: 80px;
  }

  @media (max-width: 599px) {
    > div > div:first-child {
      margin-top: 20px;
    }
  }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "basename": "",
        "children": <React.Fragment>
          <GlobalStyleComponent />
          <Route
            component={[Function]}
            path="/:activeItem?"
          />
          <ForwardRef>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/hacks"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/blog"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
              />
            </Switch>
          </ForwardRef>
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GlobalStyleComponent />,
            <Route
              component={[Function]}
              path="/:activeItem?"
            />,
            <ForwardRef>
              <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/hacks"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/blog"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                />
              </Switch>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/:activeItem?",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Switch>
                <Route
                  component={[Function]}
                  exact={true}
                  path="/hacks"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/blog"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />
                <Route
                  component={[Function]}
                />
              </Switch>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/hacks"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/blog"
                  />,
                  <Route
                    component={[Function]}
                    exact={true}
                    path="/"
                  />,
                  <Route
                    component={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/hacks",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/blog",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "exact": true,
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "rules": Array [
                  "
  display: flex;
  flex: 1;

  > div:first-child {
    display: flex;
    flex-direction: column;
    width: 100%;
  }

  > div > div:last-child {
    margin-bottom: 80px;
  }

  @media (max-width: 599px) {
    > div > div:first-child {
      margin-top: 20px;
    }
  }
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
